### ISSUE TITLE
getting error in plotting prediction of autoencodetrs

### ISSUE THREAD

**Issue by AjWattoo** [learnml-today/object-detection-with-pytorch#22]
"""i am trying to plot the prediction  """
model.eval()
for i, test_image in enumerate(test_data_loader,0):
    final, labels = test_image
    #final = final.view(1,-1)
    pred = model(final)
    _, predicted = torch.max(pred.data, 1)

plt.imshow(pred[0][0])

![1](https://user-images.githubusercontent.com/46213497/72566942-17567700-38b5-11ea-918c-64a7ad5b810b.PNG)

bcz in keras implementation ( https://blog.keras.io/building-autoencoders-in-keras.html) they just doing "**decoded_imgs = autoencoder.predict(x_test)** " and then visualizing  prediction .
for pytroch we have to do any post processing  to visualize?

**Comment by creotiv** [comment]
The result of session run in TF/Keras is not a Tensor, in PyTorch, on the other hand, every operation will give output tensor.  So to convert tensor to numpy array, you should run:
`predicted.cpu().detach().numpy()`
cpu() - move data from gpu to cpu
detach() - make tensor not backpropagated
numpy() - converts tensor to Numpy array

**Comment by AjWattoo** [comment]
worked (y)
thnx alot
